pkgs/by-name/sh/shadercross/package.nix

```diff
@@ -0,0 +1,86 @@
+{ lib
+, sdl3
+, cmake
+, ninja
+, python3
+, git
+, stdenv
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "shadercross";
+  version = "3e572c3219ea438bff849cebea34f3aad7e1859b";
+
+  outputs = [
+    "out"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "libsdl-org";
+    repo = "SDL_shadercross";
+    rev = "3e572c3219ea438bff849cebea34f3aad7e1859b";
+    hash = "sha256-m1A6JrNKLNXQWEBXYJOK7eQE1+BhhJIqOiKGrRrLxi0=";
+    fetchSubmodules = true;
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    python3
+    git
+  ];
+
+  buildInputs = [
+    sdl3
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "SDLSHADERCROSS_STATIC" true)
+    (lib.cmakeBool "SDLSHADERCROSS_CLI" true)
+    (lib.cmakeBool "SDLSHADERCROSS_DXC" true)
+    (lib.cmakeBool "SDLSHADERCROSS_VENDORED" true)
+    (lib.cmakeBool "SDLSHADERCROSS_TESTS_TRACKMEM" true)
+    (lib.cmakeBool "SDLSHADERCROSS_INSTALL" true)
+    (lib.cmakeBool "SDLSHADERCROSS_INSTALL_RUNTIME" false)
+  ];
+
+
+  doCheck = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    # Run CMake's install to place files in $out
+    DESTDIR= cmake --install . --prefix=$out # <--- Needed
+
+    runHook postInstall
```

Comment:
`postInstall` hooks should be run as the last installation step, not in the middle.